lib/util/state.js

```diff
@@ -119,16 +119,32 @@ export const getActiveItineraries = createSelector(
   }
 )
 
-/**
- * If there are no itineraries, generate an error
- * so we're not left with an empty results container.
- */
-export const createNoItinerariesErr = (state, errors, pending) => {
+export function getActiveSearchErrors(state) {
+  const activeSearch = getActiveSearch(state)
+  if (!activeSearch) return {}
+  const routingErrors = activeSearch?.response?.reduce((acc, cur) => {
+    const { routingErrors } = cur?.plan || {}
+    if (routingErrors) {
+      routingErrors.forEach((routingError) => {
+        const { code, inputField } = routingError
+        if (!acc[code]) {
+          acc[code] = new Set()
+        }
+        acc[code].add(inputField)
+      })
+    }
+    return acc
+  }, {})
   const itineraries = getActiveItineraries(state)
-  if (errors && itineraries.length === 0 && !pending) {
-    errors.NO_TRANSIT_CONNECTION = new Set()
+  const additionalErrors = {}
+  if (itineraries.length === 0 && !activeSearch?.pending) {
+    additionalErrors.NO_TRANSIT_CONNECTION = new Set()
+  }
+  if (activeSearch?.response?.some((res) => res.error)) {
+    additionalErrors.SYSTEM_ERROR = new Set()
   }
-  return errors
+
+  return { ...routingErrors, ...additionalErrors }
```

Comment:
I like your thinking here but doesn't this open the door to duplicates?

Reply:
No duplicates because it's an object with keys!

Reply:
Whoops duh!